src/ast/mod.rs

```
@@ -1,47 +1,124 @@
 use std::str;
 
-use crate::parse::{Parse, ParseError};
+use crate::parse::{self, Parse, ParseError};
+use crate::env::{EnvIter, EnvVar};
 pub mod builtin;
 mod invoke;
 
 pub use self::builtin::Builtin;
 pub use self::invoke::Invoke;
 
+#[derive(Debug, Clone)]
+pub struct ArgsIter<'a> {
+    text: &'a str
+}
+
+impl<'a> std::iter::Iterator for ArgsIter<'a> {
+    type Item = &'a str;
```

I think we're gonna want this to become an iterator of `Result<&'a str, SomeKindOfError>` eventually. Or, `next()` should halt the iteration and return none --- as it does currently --- but format the parse error first? Idk. But we need to stop swallowing errors soon.

---

(extremely C voice) set a module-global error before returning None

---

no. fuck you.